### PR TITLE
Dynamic Library

### PR DESCRIPTION
**Pull request by miguelgaio**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/186%23issuecomment-138613161%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23issuecomment-138821796%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23issuecomment-138893654%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23issuecomment-138939712%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23issuecomment-138939869%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39036025%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39037099%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39037739%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39041268%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39044930%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39044932%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39044934%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39044935%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39047172%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39047224%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39047944%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39047946%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/186%23issuecomment-138613161%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Add%20dynamic%20library%20OS%20abstraction%22%2C%20%22created_at%22%3A%20%222015-09-08T16%3A05%3A43Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40krocard%20%40cc6565%20%40OznOg%20%40dawagner%20%20%40louizo%5Cr%5Cnplease%20review%22%2C%20%22created_at%22%3A%20%222015-09-09T08%3A20%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222015-09-09T12%3A25%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%20Merge%20after%20%40dawagner%20review%22%2C%20%22created_at%22%3A%20%222015-09-09T15%3A04%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-09-09T15%3A05%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20fb87c9678ef23e8d95b3587b31b4954d60b999a6%20utility/DynamicLibrary.hpp%2044%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39036025%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22provided%22%2C%20%22created_at%22%3A%20%222015-09-09T12%3A32%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-09T13%3A58%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20utility/DynamicLibrary.hpp%3AL1-98%22%7D%2C%20%22Pull%203f1cabff6f3a713616d7027b76908dd32b3c5f1f%20utility/windows/DynamicLibrary.cpp%2064%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39041268%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22How%20do%20you%20know%20that%20%60HINSTANCE%60%20can%20be%20cast%20bijectively%20to%20%60void%20%2A%60%20%3F%5Cr%5CnMaybe%20add%20a%20static_assert%20that%20they%20are%20the%20same%20size%20%3F%22%2C%20%22created_at%22%3A%20%222015-09-09T13%3A28%3A15Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-09T13%3A58%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20utility/windows/DynamicLibrary.cpp%3AL1-75%22%7D%2C%20%22Pull%20fb87c9678ef23e8d95b3587b31b4954d60b999a6%20utility/CMakeLists.txt%2025%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39037739%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20header%20is%20private%2C%20neither%20the%20plugins%2C%20nor%20the%20client%20nedds%20it.%22%2C%20%22created_at%22%3A%20%222015-09-09T12%3A53%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-09T13%3A58%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20utility/CMakeLists.txt%3AL50-56%22%7D%2C%20%22Pull%20beb67e4e223220fc8737d582aedeb88723965da2%20utility/DynamicLibrary.hpp%2042%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39047224%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Very%20useful%2C%20I%20would%20never%20have%20%20guessed%20it%20%3B%29%22%2C%20%22created_at%22%3A%20%222015-09-09T14%3A15%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-09T14%3A21%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20utility/DynamicLibrary.hpp%3AL1-94%22%7D%2C%20%22Pull%20fb87c9678ef23e8d95b3587b31b4954d60b999a6%20parameter/SystemClass.cpp%2028%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39037099%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Proposition%3A%5Cr%5Cn%60%60%60%20cpp%5Cr%5Cntry%20%7B%5Cr%5Cn%20%20%20%20auto%20library%20%3D%20make_unique%3CDynamicLibrary%3E%28strPluginName%29%3B%5Cr%5Cn%5Cr%5Cn%20%20%20%20auto%20pluginEntry%20%3D%20getPluginSymbol%28strPluginFileName%29%3B%5Cr%5Cn%20%20%20%20//%20Make%20getSymbol%20throw%5Cr%5Cn%20%20%20%20auto%20registerBuilders%20%3D%20library.getSymbol%3CGetSubsystemBuilder%3E%28pluginEntry%29%3B%5Cr%5Cn%20%20%20%20registerBuilders%28_pSubsystemLibrary%2C%20_logger%29%3B%5Cr%5Cn%5Cr%5Cn%20%20%20%20_subsystemLibraryHandleList.push_back%28std%3A%3Amove%28library%29%29%3B%5Cr%5Cn%5Cr%5Cn%7D%20catch%20%28std%3A%3Aexception%26%20e%29%20%7B%5Cr%5Cn%20%20%20%20errors.push_back%28e.what%28%29%29%3B%5Cr%5Cn%20%20%20%20%2B%2Bit%3B%5Cr%5Cn%20%20%20%20continue%3B%5Cr%5Cn%7D%5Cr%5Cn%60%60%60%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222015-09-09T12%3A45%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-09T13%3A58%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/SystemClass.cpp%3AL211-235%22%7D%2C%20%22Pull%20beb67e4e223220fc8737d582aedeb88723965da2%20utility/DynamicLibrary.hpp%2090%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/186%23discussion_r39047172%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Grouping%20in%20doxgen%20is%20acheive%20by%20%40%7B%3A%5Cr%5Cn%60%60%60%20cpp%5Cr%5Cn/%2A%2A%20OS%20Specific%20library%20name%20patterns%5Cr%5Cn%20%2A%20%40%7B%5Cr%5Cn%20%2A/%5Cr%5Cnstatic%20const%20std%3A%3Astring%20_osLibraryPrefix%3B%5Cr%5Cnstatic%20const%20std%3A%3Astring%20_osLibrarySuffix%3B%5Cr%5Cn/%2A%2A%20%40%7D%2A/%5Cr%5Cn%60%60%60%5Cr%5Cn%5Cr%5Cnhttp%3A//www.stack.nl/%7Edimitri/doxygen/manual/grouping.html%23memgroup%22%2C%20%22created_at%22%3A%20%222015-09-09T14%3A15%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-09-09T14%3A21%3A05Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/12096004%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/miguelgaio%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20utility/DynamicLibrary.hpp%3AL1-94%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/dawagner%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%7D%2C%20%22https%3A//github.com/krocard%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/dawagner'></a><a href='https://github.com/krocard'></a>

- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/186#issuecomment-138613161'>General Comment</a></b>
- <a href='https://github.com/miguelgaio'></a> Add dynamic library OS abstraction
- <a href='https://github.com/miguelgaio'></a> @krocard @cc6565 @OznOg @dawagner  @louizo
please review
- <a href='https://github.com/OznOg'></a> :+1:
- <a href='https://github.com/krocard'></a> :shipit: Merge after @dawagner review
- [x] <a href='#crh-comment-Pull fb87c9678ef23e8d95b3587b31b4954d60b999a6 utility/DynamicLibrary.hpp 44'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/186#discussion_r39036025'>File: utility/DynamicLibrary.hpp:L1-98</a></b>
- <a href='https://github.com/krocard'></a> provided
- [x] <a href='#crh-comment-Pull fb87c9678ef23e8d95b3587b31b4954d60b999a6 parameter/SystemClass.cpp 28'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/186#discussion_r39037099'>File: parameter/SystemClass.cpp:L211-235</a></b>
- <a href='https://github.com/krocard'></a> Proposition:
``` cpp
try {
auto library = make_unique<DynamicLibrary>(strPluginName);
auto pluginEntry = getPluginSymbol(strPluginFileName);
// Make getSymbol throw
auto registerBuilders = library.getSymbol<GetSubsystemBuilder>(pluginEntry);
registerBuilders(_pSubsystemLibrary, _logger);
_subsystemLibraryHandleList.push_back(std::move(library));
} catch (std::exception& e) {
errors.push_back(e.what());
++it;
continue;
}
```
- [x] <a href='#crh-comment-Pull fb87c9678ef23e8d95b3587b31b4954d60b999a6 utility/CMakeLists.txt 25'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/186#discussion_r39037739'>File: utility/CMakeLists.txt:L50-56</a></b>
- <a href='https://github.com/krocard'></a> This header is private, neither the plugins, nor the client nedds it.
- [x] <a href='#crh-comment-Pull 3f1cabff6f3a713616d7027b76908dd32b3c5f1f utility/windows/DynamicLibrary.cpp 64'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/186#discussion_r39041268'>File: utility/windows/DynamicLibrary.cpp:L1-75</a></b>
- <a href='https://github.com/krocard'></a> How do you know that `HINSTANCE` can be cast bijectively to `void *` ?
Maybe add a static_assert that they are the same size ?
- [x] <a href='#crh-comment-Pull beb67e4e223220fc8737d582aedeb88723965da2 utility/DynamicLibrary.hpp 90'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/186#discussion_r39047172'>File: utility/DynamicLibrary.hpp:L1-94</a></b>
- <a href='https://github.com/krocard'></a> Grouping in doxgen is acheive by @{:
``` cpp
/** OS Specific library name patterns
* @{
*/
static const std::string _osLibraryPrefix;
static const std::string _osLibrarySuffix;
/** @}*/
```
http://www.stack.nl/~dimitri/doxygen/manual/grouping.html#memgroup
- [x] <a href='#crh-comment-Pull beb67e4e223220fc8737d582aedeb88723965da2 utility/DynamicLibrary.hpp 42'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/186#discussion_r39047224'>File: utility/DynamicLibrary.hpp:L1-94</a></b>
- <a href='https://github.com/krocard'></a> Very useful, I would never have  guessed it ;)

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/186?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/186?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/186?approve=0'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/186'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/186'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by miguelgaio** [comment]
Add dynamic library OS abstraction

**Comment by miguelgaio** [comment]
@krocard @cc6565 @OznOg @dawagner  @louizo
please review

**Comment by OznOg** [comment]
:+1:

File: utility/DynamicLibrary.hpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+
+
+class DynamicLibrary : private utility::NonCopyable
+{
+public:
+
+    /** Constructor
+    *
+    * @param[in] path the library path which can be privided either in absolute path or
```
Comment (krocard) [inline utility/DynamicLibrary.hpp]: provided

File: parameter/SystemClass.cpp [hunk 2]
```diff
@@ -218,31 +211,24 @@ bool CSystemClass::loadPlugins(list<string>& lstrPluginFiles, core::Results& err
         string strPluginFileName = *it;
 
         // Load attempt
-        void* lib_handle = dlopen(strPluginFileName.c_str(), RTLD_LAZY);
+        std::unique_ptr<DynamicLibrary> library;
```
Comment (krocard) [inline parameter/SystemClass.cpp]:
``` cpp
try {
    auto library = make_unique<DynamicLibrary>(strPluginName);
    // Make getSymbol throw
    auto sym = library.getSymbol<GetSubsystemBuilder>
    _subsystemLibraryHandleList.push_back(std::move(library));

} catch (std::exception& e) {
    errors.push_back(e.what());
    ++it;
    continue;
}
```

File: utility/CMakeLists.txt [hunk 3]
```diff
@@ -39,5 +50,6 @@ endif ()
 install(FILES
     NonCopyable.hpp
     Utility.h
+    DynamicLibrary.hpp
```
Comment (krocard) [inline utility/CMakeLists.txt]: This header is private, neither the plugins, nor the client nedds it.

File: utility/windows/DynamicLibrary.cpp [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+DynamicLibrary::~DynamicLibrary(void)
+{
+    HINSTANCE Module = reinterpret_cast<HINSTANCE>(_handle);
+
+    FreeLibrary(Module);
+}
+
+void *DynamicLibrary::osGetSymbol(const std::string& symbol) const
+{
+    HINSTANCE Module = reinterpret_cast<HINSTANCE>(_handle);
```
Comment (krocard) [inline utility/windows/DynamicLibrary.cpp]: How do you know that `HINSTANCE` can be cast bijectively to `void *` ?

File: parameter/SystemClass.cpp (hunk 2 above)
Comment (miguelgaio) [inline parameter/SystemClass.cpp]: :+1:<a href='#crh-update-none'></a>

File: utility/DynamicLibrary.hpp (hunk 1 above)
Comment (miguelgaio) [inline utility/DynamicLibrary.hpp]: :+1:<a href='#crh-update-none'></a>

File: utility/CMakeLists.txt (hunk 3 above)
Comment (miguelgaio) [inline utility/CMakeLists.txt]: :+1:<a href='#crh-update-none'></a>

File: utility/windows/DynamicLibrary.cpp (hunk 4 above)
Comment (miguelgaio) [inline utility/windows/DynamicLibrary.cpp]: :+1:<a href='#crh-update-complete'></a>

File: utility/DynamicLibrary.hpp [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    static std::string osSanitizePathName(const std::string& path);
+
+    /**
+    * Opaque object for library handling
+    */
+    void *_handle;
+
+    /**
+    * OS Specific library name patterns
+    */
```
Comment (krocard) [inline utility/DynamicLibrary.hpp]: Grouping in doxgen is acheive by @{:
``` cpp
/** OS Specific library name patterns
 * @{
 */
static const std::string _osLibraryPrefix;
static const std::string _osLibrarySuffix;
/** @}*/
```

http://www.stack.nl/~dimitri/doxygen/manual/grouping.html#memgroup

File: utility/DynamicLibrary.hpp [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+#include <string>
+
+
+
+class DynamicLibrary : private utility::NonCopyable
+{
+public:
+
+    /** Constructor
```
Comment (krocard) [inline utility/DynamicLibrary.hpp]: very usefull.

Comment (miguelgaio) [inline utility/DynamicLibrary.hpp]: :+1:<a href='#crh-update-none'></a>

File: utility/DynamicLibrary.hpp (hunk 5 above)
Comment (miguelgaio) [inline utility/DynamicLibrary.hpp]: :+1:<a href='#crh-update-complete'></a>

**Comment by krocard** [comment]
:shipit:

**Comment by dawagner** [comment]
:shipit:

**Pull request merged**